pkg/planner/core/stats.go

```diff
@@ -404,21 +408,24 @@ func detachCondAndBuildRangeForPath(
 	return err
 }
 
-func getMinSelectivityFromPaths(paths []*util.AccessPath, totalRowCount float64) float64 {
+func getGeneralAttributesFromPaths(paths []*util.AccessPath, totalRowCount float64) (float64, bool) {
 	minSelectivity := 1.0
-	if totalRowCount <= 0 {
-		return minSelectivity
-	}
+	hasForce := false
 	for _, path := range paths {
 		// For table path and index merge path, AccessPath.CountAfterIndex is not set and meaningless,
 		// but we still consider their AccessPath.CountAfterAccess.
-		if path.IsTablePath() || path.PartialIndexPaths != nil {
-			minSelectivity = min(minSelectivity, path.CountAfterAccess/totalRowCount)
-			continue
+		if totalRowCount > 0 {
+			if path.IsTablePath() || path.PartialIndexPaths != nil {
+				minSelectivity = min(minSelectivity, path.CountAfterAccess/totalRowCount)
+			} else {
+				minSelectivity = min(minSelectivity, path.CountAfterIndex/totalRowCount)
+			}
+		}
+		if !hasForce && path.Forced {
```

Comment:
User can use "use index" to force table scan like: 
```
create table t (a int, b int, primary key(a), key(b));
select /*+ use_index(t, primary) */ * from t;
```
Should we exclude this case like `!path.IsTablePath()`?

Reply:
Or our assumption here is that if the user is using `use_index` hint, they prefer `range scan` no matter it's table-range-scan or index-range-scan?